### PR TITLE
Fix generation scripts

### LINKED ISSUE #155: Pihole won't start because *.hosts are placed in the "/etc/dnsmasq.d" directory
### Describe the issue you are having
*.hosts files should not be placed in the `/etc/dnsmasq.d` directory, because dnsmasq interprets anything in that directory as a configuration file. I don't know if this breaks _lancachenet/lancache-dns_ but it does break pihole, which crashes on startup when dnsmasq fails to parse those files.

The solution is to instead place the *.hosts files into `/etc/dnsmasq.hosts` (or whatever) instead. Half of this solution is contained in [this commit](https://github.com/DragonQ/cache-domains/commit/3e155a7afc883de4caafd3c9cc54d12c69eae9dd), which makes `create-dnsmasq.sh` change the location of the *hosts files written in the config files. The other half of the solution is to ensure the *.hosts files are actually located in that new directory. I use a script to do this:

```
./create-dnsmasq.sh
sudo cp -rf ./output/dnsmasq/*.conf ${DNSMASQ_DIR}
sudo cp -rf ./output/dnsmasq/*.hosts ${HOSTS_DIR}
```

I am unsure what changes would be needed to ensure compatibility with _lancachenet/lancache-dns_, however. Presumably they have some kind of script that pulls the files from the `output` directory to wherever they need to be, in which case that would need updating to match.

### Describe your setup?
I have a pihole container with its dnsmasq directories populated with the files generated by the `cache-domains` script. This ensures any compatible game servers are redirected to the lancache/monolithic container, and all other traffic is routed to a cloudflared container.

### Are you running sniproxy
Yes

### DNS Configuration
I use this docker-compose script for pi-hole, lancache, and DNSSEC:

```
version: '3.7'
services:
    lancache-monolithic:
        container_name: lancache
        image: 'lancachenet/monolithic:latest'
        restart: unless-stopped
        networks:
            pihole-net:
                ipv4_address: ${LANCACHE_IP}
        volumes:
            - '/data/cache:/data/cache'
            - '/data/logs:/data/logs'
        ports:
            - '${HOST_IP}:80:80'
        environment:
            - UPSTREAM_DNS=${DNS_SERVERS}
            - CACHE_MEM_SIZE=${CACHE_MEM_SIZE}
            - CACHE_DISK_SIZE=${CACHE_DISK_SIZE}
            - CACHE_DOMAINS_REPO=${CACHE_DOMAINS_REPO}
            - CACHE_DOMAINS_BRANCH=${CACHE_DOMAINS_BRANCH}
    sniproxy:
        container_name: lancache-sniproxy
        image: 'lancachenet/sniproxy:latest'
        restart: unless-stopped
        networks:
            pihole-net:
                ipv4_address: ${SNIPROXY_IP}
        ports:
            - '${HOST_IP}:443:443'
        environment:
            - UPSTREAM_DNS=${DNS_SERVERS_SNI}
    pihole:
        container_name: pihole
        image: 'pihole/pihole:latest'
        restart: unless-stopped
        networks:
            pihole-net:
                ipv4_address: ${PIHOLE_IP}
        volumes:
            - ${PIHOLE_ROOT_PATH}/etc/:/etc/pihole/
            - ${PIHOLE_ROOT_PATH}/dnsmasq.d/:/etc/dnsmasq.d/
            - ${PIHOLE_ROOT_PATH}/dnsmasq.hosts/:/etc/dnsmasq.hosts/
        ports:
            - '${HOST_IP}:53:53/tcp'
            - '${HOST_IP}:53:53/udp'
            - '${HOST_IP}:8080:80/tcp'
        environment:
            - TZ='Europe/London'
            - WEBPASSWORD=${WEB_PASSWORD}
            - DNS1=${CLOUDFLARED_DNS_IP}
            - DNS2=no
        dns:
            - 127.0.0.1
            - 1.1.1.1
    cloudflared:
        container_name: cloudflared
        image: 'crazymax/cloudflared:latest'
        restart: unless-stopped
        networks:
            pihole-net:
                ipv4_address: ${CLOUDFLARED_IP}
        ports:
            #- '${HOST_IP}:53:5053/udp'
            - '${HOST_IP}:49312:49312/tcp'
        environment:
            - "TZ=Europe/London"
            - "TUNNEL_DNS_UPSTREAM=${UPSTREAM_DNS_SERVERS}"
    ouroboros:
        container_name: ouroboros
        hostname: ouroboros
        image: pyouroboros/ouroboros
        restart: unless-stopped
        networks:
            pihole-net:
                ipv4_address: ${OUROBOROS_IP}
        volumes:
            - /var/run/docker.sock:/var/run/docker.sock
        environment:
            - CLEANUP=true
            - INTERVAL=86400
            - CRON="5 1 * * *"
            - LOG_LEVEL=info
            - SELF_UPDATE=true
            - IGNORE="mongo influxdb postgres mariadb"
            - TZ=Europe/London

networks:
  pihole-net:
    ipam:
      config:
        - subnet: ${IP_SUBNET}
```

### Sniproxy output
Please paste the output from `docker logs <sniproxy container name/id> | sed 's/.*\:443 \[//;s/\].*//' | sort | uniq -c` below
```
      1 cdn1.unrealengine.com
      1 cdn2.unrealengine.com
      1 cm2-ams1.cm.steampowered.com
      1 cm3-lhr1.cm.steampowered.com
      5 cp601.prod.do.dsp.mp.microsoft.com
      3 geo.prod.do.dsp.mp.microsoft.com
      3 kv601.prod.do.dsp.mp.microsoft.com
      7 nydus.battle.net
      1 static8.cdn.ubi.com
     10 steamcdn-a.akamaihd.net
      1 ubiservices.cdn.ubi.com
```

### PR DESCRIPTION
**Pull request by nightah**
This change leverages #130 and also applies this to the dnsmasq script.

As it currently stands both generation scripts (unbound and dnsmasq) have a condition where a domain will be skipped if it fuzzy matches a domain already parsed that is higher in the CDN domain list.

For example the latter of the below two samples would never be added.
https://github.com/uklans/cache-domains/blob/8793ce15315cac1e594f7602158c2e82f510bc91/steam.txt#L20 https://github.com/uklans/cache-domains/blob/8793ce15315cac1e594f7602158c2e82f510bc91/steam.txt#L29

I've also taken the liberty to sort the output of said scripts for readability and troubleshooting purposes.

Example outputs of each script before this commit and after are provided below.

For context the Steam CDN currently has 51 entries of which 9 are wildcards.
dnsmasq needs to cater for wildcards in the `lancache.conf` and all other hosts in the respective CDN's output `.hosts` file, so this should be a 9 (wildcard) + 42 (non-wildcard) host split:

**[dnsmasq] `lancache.conf` before:
(9 unsorted wildcard entries and .hosts ref)**
```conf
addn-hosts=/etc/dnsmasq.d/steam.hosts
address=/content.steampowered.com/10.10.10.200
address=/hsar.steampowered.com.edgesuite.net/10.10.10.200
address=/akamai.steamstatic.com/10.10.10.200
address=/cs.steampowered.com/10.10.10.200
address=/cm.steampowered.com/10.10.10.200
address=/edgecast.steamstatic.com/10.10.10.200
address=/steamcontent.com/10.10.10.200
address=/steam-content-dnld-1.apac-1-cdn.cqloud.com/10.10.10.200
address=/steam-content-dnld-1.eu-c1-cdn.cqloud.com/10.10.10.200
```

**[dnsmasq] `lancache.conf` after:
(9 sorted wildcard entries and .hosts ref)**
```conf
addn-hosts=/etc/dnsmasq.d/steam.hosts
address=/akamai.steamstatic.com/10.10.10.200
address=/cm.steampowered.com/10.10.10.200
address=/content.steampowered.com/10.10.10.200
address=/cs.steampowered.com/10.10.10.200
address=/edgecast.steamstatic.com/10.10.10.200
address=/hsar.steampowered.com.edgesuite.net/10.10.10.200
address=/steamcontent.com/10.10.10.200
address=/steam-content-dnld-1.apac-1-cdn.cqloud.com/10.10.10.200
address=/steam-content-dnld-1.eu-c1-cdn.cqloud.com/10.10.10.200
```

**[dnsmasq] `steam.hosts` before:
(46 unsorted entries, missing fuzzy matches and not properly excluding wildcards)**
```hosts
10.10.10.200 lancache.steamcontent.com
10.10.10.200 content.steampowered.com
10.10.10.200 content1.steampowered.com
10.10.10.200 content2.steampowered.com
10.10.10.200 content3.steampowered.com
10.10.10.200 content4.steampowered.com
10.10.10.200 content5.steampowered.com
10.10.10.200 content6.steampowered.com
10.10.10.200 content7.steampowered.com
10.10.10.200 content8.steampowered.com
10.10.10.200 cs.steampowered.com
10.10.10.200 client-download.steampowered.com
10.10.10.200 hsar.steampowered.com.edgesuite.net
10.10.10.200 akamai.steamstatic.com
10.10.10.200 content-origin.steampowered.com
10.10.10.200 clientconfig.akamai.steamtransparent.com
10.10.10.200 steampipe.akamaized.net
10.10.10.200 edgecast.steamstatic.com
10.10.10.200 steam.apac.qtlglb.com.mwcloudcdn.com
10.10.10.200 cm.steampowered.com
10.10.10.200 cdn1-sea1.valve.net
10.10.10.200 cdn2-sea1.valve.net
10.10.10.200 steam-content-dnld-1.apac-1-cdn.cqloud.com
10.10.10.200 steam-content-dnld-1.eu-c1-cdn.cqloud.com
10.10.10.200 edge.steam-dns.top.comcast.net
10.10.10.200 edge.steam-dns-2.top.comcast.net
10.10.10.200 steam.naeu.qtlglb.com
10.10.10.200 steampipe-kr.akamaized.net
10.10.10.200 steam.ix.asn.au
10.10.10.200 steam.eca.qtlglb.com
10.10.10.200 steam.cdn.on.net
10.10.10.200 update5.dota2.wmsj.cn
10.10.10.200 update2.dota2.wmsj.cn
10.10.10.200 update6.dota2.wmsj.cn
10.10.10.200 update3.dota2.wmsj.cn
10.10.10.200 update1.dota2.wmsj.cn
10.10.10.200 update4.dota2.wmsj.cn
10.10.10.200 update5.csgo.wmsj.cn
10.10.10.200 update2.csgo.wmsj.cn
10.10.10.200 update4.csgo.wmsj.cn
10.10.10.200 update3.csgo.wmsj.cn
10.10.10.200 update6.csgo.wmsj.cn
10.10.10.200 update1.csgo.wmsj.cn
10.10.10.200 st.dl.bscstorage.net
10.10.10.200 cdn.mileweb.cs.steampowered.com.8686c.com
10.10.10.200 steamcdn-a.akamaihd.net
```

**[dnsmasq] `steam.hosts` after:
(42 sorted entries, only includes non-wildcard entries as intended and picks up fuzzy matches)**
```hosts
10.10.10.200 cdn1-sea1.valve.net
10.10.10.200 cdn2-sea1.valve.net
10.10.10.200 cdn.mileweb.cs.steampowered.com.8686c.com
10.10.10.200 clientconfig.akamai.steamtransparent.com
10.10.10.200 client-download.steampowered.com
10.10.10.200 content1.steampowered.com
10.10.10.200 content2.steampowered.com
10.10.10.200 content3.steampowered.com
10.10.10.200 content4.steampowered.com
10.10.10.200 content5.steampowered.com
10.10.10.200 content6.steampowered.com
10.10.10.200 content7.steampowered.com
10.10.10.200 content8.steampowered.com
10.10.10.200 content-origin.steampowered.com
10.10.10.200 cs.steampowered.com
10.10.10.200 edgecast.steamstatic.com
10.10.10.200 edge.steam-dns-2.top.comcast.net
10.10.10.200 edge.steam-dns.top.comcast.net
10.10.10.200 lancache.steamcontent.com
10.10.10.200 st.dl.bscstorage.net
10.10.10.200 steam.apac.qtlglb.com
10.10.10.200 steam.apac.qtlglb.com.mwcloudcdn.com
10.10.10.200 steamcdn-a.akamaihd.net
10.10.10.200 steam.cdn.on.net
10.10.10.200 steamcontent.com
10.10.10.200 steam.eca.qtlglb.com
10.10.10.200 steam.ix.asn.au
10.10.10.200 steam.naeu.qtlglb.com
10.10.10.200 steampipe.akamaized.net
10.10.10.200 steampipe-kr.akamaized.net
10.10.10.200 update1.csgo.wmsj.cn
10.10.10.200 update1.dota2.wmsj.cn
10.10.10.200 update2.csgo.wmsj.cn
10.10.10.200 update2.dota2.wmsj.cn
10.10.10.200 update3.csgo.wmsj.cn
10.10.10.200 update3.dota2.wmsj.cn
10.10.10.200 update4.csgo.wmsj.cn
10.10.10.200 update4.dota2.wmsj.cn
10.10.10.200 update5.csgo.wmsj.cn
10.10.10.200 update5.dota2.wmsj.cn
10.10.10.200 update6.csgo.wmsj.cn
10.10.10.200 update6.dota2.wmsj.cn
```

**[unbound] `steam.conf` before:
(46 unsorted entries, missing fuzzy matches)**
```conf
server:
  local-zone: "lancache.steamcontent.com" redirect
  local-data: "lancache.steamcontent.com 30 IN A 10.10.10.200"
  local-zone: "content.steampowered.com" redirect
  local-data: "content.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content1.steampowered.com" redirect
  local-data: "content1.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content2.steampowered.com" redirect
  local-data: "content2.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content3.steampowered.com" redirect
  local-data: "content3.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content4.steampowered.com" redirect
  local-data: "content4.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content5.steampowered.com" redirect
  local-data: "content5.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content6.steampowered.com" redirect
  local-data: "content6.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content7.steampowered.com" redirect
  local-data: "content7.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content8.steampowered.com" redirect
  local-data: "content8.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "cs.steampowered.com" redirect
  local-data: "cs.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "client-download.steampowered.com" redirect
  local-data: "client-download.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "hsar.steampowered.com.edgesuite.net" redirect
  local-data: "hsar.steampowered.com.edgesuite.net 30 IN A 10.10.10.200"
  local-zone: "akamai.steamstatic.com" redirect
  local-data: "akamai.steamstatic.com 30 IN A 10.10.10.200"
  local-zone: "content-origin.steampowered.com" redirect
  local-data: "content-origin.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "clientconfig.akamai.steamtransparent.com" redirect
  local-data: "clientconfig.akamai.steamtransparent.com 30 IN A 10.10.10.200"
  local-zone: "steampipe.akamaized.net" redirect
  local-data: "steampipe.akamaized.net 30 IN A 10.10.10.200"
  local-zone: "edgecast.steamstatic.com" redirect
  local-data: "edgecast.steamstatic.com 30 IN A 10.10.10.200"
  local-zone: "steam.apac.qtlglb.com.mwcloudcdn.com" redirect
  local-data: "steam.apac.qtlglb.com.mwcloudcdn.com 30 IN A 10.10.10.200"
  local-zone: "cm.steampowered.com" redirect
  local-data: "cm.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "cdn1-sea1.valve.net" redirect
  local-data: "cdn1-sea1.valve.net 30 IN A 10.10.10.200"
  local-zone: "cdn2-sea1.valve.net" redirect
  local-data: "cdn2-sea1.valve.net 30 IN A 10.10.10.200"
  local-zone: "steam-content-dnld-1.apac-1-cdn.cqloud.com" redirect
  local-data: "steam-content-dnld-1.apac-1-cdn.cqloud.com 30 IN A 10.10.10.200"
  local-zone: "steam-content-dnld-1.eu-c1-cdn.cqloud.com" redirect
  local-data: "steam-content-dnld-1.eu-c1-cdn.cqloud.com 30 IN A 10.10.10.200"
  local-zone: "edge.steam-dns.top.comcast.net" redirect
  local-data: "edge.steam-dns.top.comcast.net 30 IN A 10.10.10.200"
  local-zone: "edge.steam-dns-2.top.comcast.net" redirect
  local-data: "edge.steam-dns-2.top.comcast.net 30 IN A 10.10.10.200"
  local-zone: "steam.naeu.qtlglb.com" redirect
  local-data: "steam.naeu.qtlglb.com 30 IN A 10.10.10.200"
  local-zone: "steampipe-kr.akamaized.net" redirect
  local-data: "steampipe-kr.akamaized.net 30 IN A 10.10.10.200"
  local-zone: "steam.ix.asn.au" redirect
  local-data: "steam.ix.asn.au 30 IN A 10.10.10.200"
  local-zone: "steam.eca.qtlglb.com" redirect
  local-data: "steam.eca.qtlglb.com 30 IN A 10.10.10.200"
  local-zone: "steam.cdn.on.net" redirect
  local-data: "steam.cdn.on.net 30 IN A 10.10.10.200"
  local-zone: "update5.dota2.wmsj.cn" redirect
  local-data: "update5.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update2.dota2.wmsj.cn" redirect
  local-data: "update2.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update6.dota2.wmsj.cn" redirect
  local-data: "update6.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update3.dota2.wmsj.cn" redirect
  local-data: "update3.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update1.dota2.wmsj.cn" redirect
  local-data: "update1.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update4.dota2.wmsj.cn" redirect
  local-data: "update4.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update5.csgo.wmsj.cn" redirect
  local-data: "update5.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update2.csgo.wmsj.cn" redirect
  local-data: "update2.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update4.csgo.wmsj.cn" redirect
  local-data: "update4.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update3.csgo.wmsj.cn" redirect
  local-data: "update3.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update6.csgo.wmsj.cn" redirect
  local-data: "update6.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update1.csgo.wmsj.cn" redirect
  local-data: "update1.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "st.dl.bscstorage.net" redirect
  local-data: "st.dl.bscstorage.net 30 IN A 10.10.10.200"
  local-zone: "cdn.mileweb.cs.steampowered.com.8686c.com" redirect
  local-data: "cdn.mileweb.cs.steampowered.com.8686c.com 30 IN A 10.10.10.200"
  local-zone: "steamcdn-a.akamaihd.net" redirect
  local-data: "steamcdn-a.akamaihd.net 30 IN A 10.10.10.200"
```

**[unbound] `steam.conf` after:
(51 sorted entries, picks up fuzzy matches)**
```hosts
server:
  local-zone: "akamai.steamstatic.com" redirect
  local-data: "akamai.steamstatic.com 30 IN A 10.10.10.200"
  local-zone: "cdn1-sea1.valve.net" redirect
  local-data: "cdn1-sea1.valve.net 30 IN A 10.10.10.200"
  local-zone: "cdn2-sea1.valve.net" redirect
  local-data: "cdn2-sea1.valve.net 30 IN A 10.10.10.200"
  local-zone: "cdn.mileweb.cs.steampowered.com.8686c.com" redirect
  local-data: "cdn.mileweb.cs.steampowered.com.8686c.com 30 IN A 10.10.10.200"
  local-zone: "clientconfig.akamai.steamtransparent.com" redirect
  local-data: "clientconfig.akamai.steamtransparent.com 30 IN A 10.10.10.200"
  local-zone: "client-download.steampowered.com" redirect
  local-data: "client-download.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "cm.steampowered.com" redirect
  local-data: "cm.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content1.steampowered.com" redirect
  local-data: "content1.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content2.steampowered.com" redirect
  local-data: "content2.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content3.steampowered.com" redirect
  local-data: "content3.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content4.steampowered.com" redirect
  local-data: "content4.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content5.steampowered.com" redirect
  local-data: "content5.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content6.steampowered.com" redirect
  local-data: "content6.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content7.steampowered.com" redirect
  local-data: "content7.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content8.steampowered.com" redirect
  local-data: "content8.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content-origin.steampowered.com" redirect
  local-data: "content-origin.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "content.steampowered.com" redirect
  local-data: "content.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "cs.steampowered.com" redirect
  local-data: "cs.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "cs.steampowered.com" redirect
  local-data: "cs.steampowered.com 30 IN A 10.10.10.200"
  local-zone: "edgecast.steamstatic.com" redirect
  local-data: "edgecast.steamstatic.com 30 IN A 10.10.10.200"
  local-zone: "edgecast.steamstatic.com" redirect
  local-data: "edgecast.steamstatic.com 30 IN A 10.10.10.200"
  local-zone: "edge.steam-dns-2.top.comcast.net" redirect
  local-data: "edge.steam-dns-2.top.comcast.net 30 IN A 10.10.10.200"
  local-zone: "edge.steam-dns.top.comcast.net" redirect
  local-data: "edge.steam-dns.top.comcast.net 30 IN A 10.10.10.200"
  local-zone: "hsar.steampowered.com.edgesuite.net" redirect
  local-data: "hsar.steampowered.com.edgesuite.net 30 IN A 10.10.10.200"
  local-zone: "lancache.steamcontent.com" redirect
  local-data: "lancache.steamcontent.com 30 IN A 10.10.10.200"
  local-zone: "st.dl.bscstorage.net" redirect
  local-data: "st.dl.bscstorage.net 30 IN A 10.10.10.200"
  local-zone: "steam.apac.qtlglb.com" redirect
  local-data: "steam.apac.qtlglb.com 30 IN A 10.10.10.200"
  local-zone: "steam.apac.qtlglb.com.mwcloudcdn.com" redirect
  local-data: "steam.apac.qtlglb.com.mwcloudcdn.com 30 IN A 10.10.10.200"
  local-zone: "steamcdn-a.akamaihd.net" redirect
  local-data: "steamcdn-a.akamaihd.net 30 IN A 10.10.10.200"
  local-zone: "steam.cdn.on.net" redirect
  local-data: "steam.cdn.on.net 30 IN A 10.10.10.200"
  local-zone: "steamcontent.com" redirect
  local-data: "steamcontent.com 30 IN A 10.10.10.200"
  local-zone: "steamcontent.com" redirect
  local-data: "steamcontent.com 30 IN A 10.10.10.200"
  local-zone: "steam-content-dnld-1.apac-1-cdn.cqloud.com" redirect
  local-data: "steam-content-dnld-1.apac-1-cdn.cqloud.com 30 IN A 10.10.10.200"
  local-zone: "steam-content-dnld-1.eu-c1-cdn.cqloud.com" redirect
  local-data: "steam-content-dnld-1.eu-c1-cdn.cqloud.com 30 IN A 10.10.10.200"
  local-zone: "steam.eca.qtlglb.com" redirect
  local-data: "steam.eca.qtlglb.com 30 IN A 10.10.10.200"
  local-zone: "steam.ix.asn.au" redirect
  local-data: "steam.ix.asn.au 30 IN A 10.10.10.200"
  local-zone: "steam.naeu.qtlglb.com" redirect
  local-data: "steam.naeu.qtlglb.com 30 IN A 10.10.10.200"
  local-zone: "steampipe.akamaized.net" redirect
  local-data: "steampipe.akamaized.net 30 IN A 10.10.10.200"
  local-zone: "steampipe-kr.akamaized.net" redirect
  local-data: "steampipe-kr.akamaized.net 30 IN A 10.10.10.200"
  local-zone: "update1.csgo.wmsj.cn" redirect
  local-data: "update1.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update1.dota2.wmsj.cn" redirect
  local-data: "update1.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update2.csgo.wmsj.cn" redirect
  local-data: "update2.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update2.dota2.wmsj.cn" redirect
  local-data: "update2.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update3.csgo.wmsj.cn" redirect
  local-data: "update3.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update3.dota2.wmsj.cn" redirect
  local-data: "update3.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update4.csgo.wmsj.cn" redirect
  local-data: "update4.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update4.dota2.wmsj.cn" redirect
  local-data: "update4.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update5.csgo.wmsj.cn" redirect
  local-data: "update5.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update5.dota2.wmsj.cn" redirect
  local-data: "update5.dota2.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update6.csgo.wmsj.cn" redirect
  local-data: "update6.csgo.wmsj.cn 30 IN A 10.10.10.200"
  local-zone: "update6.dota2.wmsj.cn" redirect
  local-data: "update6.dota2.wmsj.cn 30 IN A 10.10.10.200"
```

Closes #130.
Closes #155.
Closes #156.

### CODE REVIEW

**Comment by nightah** [comment]
@mintopia any chance of a review/merge on this please?

My commentary above includes full outputs for all scenarios that are affected.

**Comment by nightah** [comment]
@VibroAxe I've added a `README.md` which hopefully provides some more information and context for users.

I also noticed a couple typos in the main `README.md` which I've also addressed in this PR.

File: scripts/README.md [hunk 1]
```diff
@@ -46,4 +46,4 @@ The `lancache.conf` should be copied into the `/etc/dnsmasq.d/` location but als
 You can copy the `*.hosts` file to any location other than `/etc/dnsmasq.d/` as this location is utilised only for `*.conf` files.
```
Comment (VibroAxe) [inline scripts/README.md:46]: `Any location other than /etc/dnsmasq/hosts/`

File: scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "riot":         "generic",
+    "sony":         "generic",
+    "steam":        "generic",
+    "uplay":        "generic",
+    "wsus":         "generic"
+  }
+}
+```
+3. Run generation script relative to your DNS implementation: `bash create-dnsmasq.sh`.
+4. Copy files from `output/{dnsmasq,unbound}/*` to the respective locations for Dnsmasq/Unbound.
```
Comment (VibroAxe) [inline scripts/README.md:35]: I don't use the scripts but is it clear in the output WHERE/WHICH files to copy. 

Perhaps something at the end which echo's to console where the files are and where the expected move should be?

File: scripts/create-dnsmasq.sh [hunk 3]
```diff
@@ -45,37 +45,37 @@ while read -r entry; do
...
                                         continue
                                 fi
                                 for i in ${cacheip}; do
                                         echo "${i} ${parsed}" >> "$outputfile"
                                 done
-                        done <<< $(cat ${basedir}/$filename);
+                        done <<< $(cat ${basedir}/$filename | sort);
                 done <<< $(jq -r ".cache_domains[$entry].domain_files[$fileid]" $path)
         done <<< $(jq -r ".cache_domains[$entry].domain_files | to_entries[] | .key" $path)
 done <<< $(jq -r '.cache_domains | to_entries[] | .key' $path)
```
Comment (VibroAxe) [inline scripts/create-dnsmasq.sh:81]: Add contextual output here back to console detailing what needs to happen next

File: scripts/create-unbound.sh [hunk 4]
```diff
@@ -52,14 +52,14 @@ while read entry; do
...
 				fi
 				echo "  local-zone: \"${parsed}\" redirect" >> $outputfile
 				for i in ${cacheip}; do
 					echo "  local-data: \"${parsed} 30 IN A ${i}\"" >> $outputfile
 				done
-			done <<< $(cat ${basedir}/$filename);
+			done <<< $(cat ${basedir}/$filename | sort);
 		done <<< $(jq -r ".cache_domains[$entry].domain_files[$fileid]" $path)
 	done <<< $(jq -r ".cache_domains[$entry].domain_files | to_entries[] | .key" $path)
 done <<< $(jq -r '.cache_domains | to_entries[] | .key' $path)
```
Comment (VibroAxe) [inline scripts/create-unbound.sh:65]: Add contextual output here back to console detailing what needs to happen next

**Review by VibroAxe — CHANGES_REQUESTED** [review]
Mostly minor changes @nightah the proposals look good, just some documentation to account for PEBKAC issues

**Comment by VibroAxe** [comment]
This now replaces #156 and will resolve #155

File: scripts/README.md (hunk 1 above)
Comment (nightah) [reply scripts/README.md:46]: That would be wrong though, `/etc/dnsmasq.d/` is where `.conf` files are intended to go and if a `.hosts` file is loaded in that directory dnsmasq will fail to start.

File: scripts/README.md (hunk 2 above)
Comment (nightah) [reply scripts/README.md:35]: Ack.

**Comment by nightah** [comment]
> This now replaces #156 and will resolve #155

I've already updated the initial comment with closes statements:

> Closes #130.
> Closes #155.
> Closes #156.

File: scripts/create-unbound.sh (hunk 4 above)
Comment (VibroAxe) [reply scripts/create-unbound.sh:65]: I don't see a commit with changes to the unbound script?

File: scripts/README.md (hunk 1 above)
Comment (VibroAxe) [reply scripts/README.md:46]: Yup, I see. Misread the intent

File: scripts/create-dnsmasq.sh [hunk 5]
```diff
@@ -79,3 +79,5 @@ while read -r entry; do
...
+
+echo "Please copy the following files:\n- ./output/dnsmasq/lancache.conf to /etc/dnsmasq/dnsmasq.d/\n- ./output/dnsmasq/hosts to /etc/dnsmasq/"
```
Comment (VibroAxe) [inline scripts/create-dnsmasq.sh:83]: Should probably use $outputdir here to make sure that this is future/change proofed

**Review by VibroAxe — APPROVED** [review]

File: .gitignore [hunk 6]
```diff
@@ -1 +1,2 @@
 scripts/output
+scripts/config.json
```
Comment (DragonQ) [inline .gitignore:2]: Is this a good change? Not sure about others but I know that I would want my custom `config.json` to be in source control.

Comment (VibroAxe) [reply .gitignore:2]: My view is it's a good change for our repo as we intentionally don't store any config beyond the example, you may wish to override it for your local repos

@nightah you use this regularly as well, thoughts?

Comment (nightah) [reply .gitignore:2]: It's to ensure that a user doesn't accidently commit their configuration file back to the main repo.
If you want to maintain your own repo and commit to said repo you can do that even with that entry in the `.gitignore`.

Comment (VibroAxe) [reply .gitignore:2]: @DragonQ fyi `git add config.json --force` will override the gitignore

**Pull request merged**